services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -2496,6 +2498,11 @@ private Pair<Boolean, Version> addVersion(
             }
             // Update ZK with the new version
             store.addVersion(version, true);
+            // update the age of the curent version to 0 for deletion in StoreBackupVersionCleanupService
+            if (isRepush && multiClusterConfigs.getControllerConfig(clusterName)
+                .isBackupVersionMetadataFetchBasedCleanupEnabled()) {
+              store.getVersion(currentVersionBeforePush).get().setCreatedTime(Duration.ZERO);
+            }
```

Comment:
This may not be fully correct in all scenarios.
Let us say:
```
1. User runs a batch push job, which produces a version: v10.
2. The job succeeds in two regions with: v9, v10.
3. The job fails in one region: v9.
4. We trigger a repush based on v10.
5. Two good regions will have: v9, v10, v11 and the bad region will have v9, v11.
6. Based on this logic, the v10 in good regions will be marked with created time: 0 and the v9 in the bad region will be marked in the same way.
7. v10 will be removed from good regions and v9 will be removed from the bad region.
```
The above doesn't sound good and is not consistent.

I think we should introduce a new version-level field called re-push source version, so that we can explicitly know which versions to cleanup.
Right now, I feel the logic is a little tricky and error prone.

Reply:
emm V9 wont be deleted in bad region as we have check for keep min number of version.

Reply:
Can you show me the logic regarding ^?
Also v9's created time will be updated to be `0` based on the new logic IIUC.
I think it is confusing, and it is better to make the logic clean to avoid tricky bugs.